src/components/Dialog/__tests__/Dialog.jest.js

```diff
@@ -0,0 +1,23 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import Dialog from "../Dialog";
+
+describe("Dialog tests", () => {
+  describe("Click outside", () => {
+    it("should call onClickOutside and close dialog", async () => {
+      const onClickOutsideMock = jest.fn();
+      const element = document.createElement("div");
+      document.body.appendChild(element);
+
+      render(<Dialog onClickOutside={onClickOutsideMock} content={<div>Dialog</div>} open />, { baseElement: element });
```

Comment:
this test only check onclickoutside function it's not test if the component hide the dialog when user click outside :/

Reply:
I check both things:
expect(onClickOutsideMock).toBeCalled();
expect(screen.findByText("Dialog")).not.toBeInTheDocument;